src/storage_node.rs

```diff
@@ -105,20 +109,15 @@ pub enum State {
     /// In this state, `t` is NOT valid, and is uninitialized. it MUST NOT be
     /// read.
     NonResident,
-    /// The value has been initialized using a default value (NOT from flash)
-    /// and needs to be written to flash.
-    ///
-    /// In this state, `t` IS valid, and may be read at any time (by the holder
-    /// of the lock).
-    DefaultUnwritten,
     /// The value has been initialized using a value from flash. No writes are
     /// pending.
     ///
     /// In this state, `t` IS valid, and may be read at any time (by the holder
     /// of the lock).
     ValidNoWriteNeeded,
     /// The value has been written, but these changes have NOT been flushed back
-    /// to the flash.
+    /// to the flash. This includes the case where the node has "written back"
+    /// the default value.
```

Comment:
```suggestion
    /// a user-defined default value.
```

We can have a custom "default" closure and don't always init with T::default.